### PR TITLE
docs: add documentation for `Tree.inspect`

### LINKED ISSUE #1759: Missing documentation in `Tree.inspect`
See https://github.com/mirage/irmin/pull/1755#discussion_r802408387

### PR DESCRIPTION
**Pull request by metanivek**
Closes #1759

### CODE REVIEW

File: src/irmin/tree.ml [hunk 1]
```diff
@@ -430,12 +430,8 @@ module Make (P : Backend.S) = struct
...
 
     and t = { mutable v : v; info : info }
-    (** [t.v] has 3 possible states:
-
-        - A [Map], only after a [Tree.of_concrete] operation.
-        - A [Value], only after an add, a remove, temporarily during an export
-          or at the end of a merge.
-        - It is otherwise a [Key].
+    (** For disussion of [t.v]'s states, see interface documentation for
+        [Tree.inspect].
```
Comment (tomjridge) [inline src/irmin/tree.ml:434]: Suggested change: `{!Tree_intf.inspect}` (or similar) i.e., directly link to the inspect function in *Tree_intf*

Comment (metanivek) [reply src/irmin/tree.ml]: :+1: good idea!

File: src/irmin/tree.ml [hunk 2]
```diff
@@ -430,12 +430,7 @@ module Make (P : Backend.S) = struct
       | Pruned of hash
 
     and t = { mutable v : v; info : info }
-    (** [t.v] has 3 possible states:
-
-        - A [Map], only after a [Tree.of_concrete] operation.
-        - A [Value], only after an add, a remove, temporarily during an export
-          or at the end of a merge.
-        - It is otherwise a [Key].
+    (** For disussion of [t.v]'s states, see {!Tree_intf.S.inspect}.
```
Comment (zshipko) [inline src/irmin/tree.ml:433]:
```suggestion
    (** For discussion of [t.v]'s states, see {!Tree_intf.S.inspect}.
```

Comment (metanivek) [reply src/irmin/tree.ml]: :pray:

**Review by Ngoguey42 — APPROVED** [review]
Thanks

File: src/irmin/tree_intf.ml [hunk 3]
```diff
@@ -391,6 +391,18 @@ module type S = sig
   val inspect :
     t ->
     [ `Contents | `Node of [ `Map | `Key | `Value | `Portable_dirty | `Pruned ] ]
+  (** [inspect t] is similar to {!kind}, with addtional state information for
```
[the remark this answers is not in the document]
Comment (metanivek) [reply src/irmin/tree_intf.ml]: :pray:

**Pull request merged**